pkg/workflow/types.go

```diff
@@ -18,6 +18,10 @@ import (
 type Identifier = *url.URL
 type Callback func(invocation InvocationContext, input []Data) ([]Data, error)
 type ExtensionInit func(engine Engine) error
+type RuntimeInfo struct {
```

Comment:
Suggestion: For the API interface it would be best to have RuntimeInfo as an interface and instead of public members, use getter and setter function. This way we are more open for future evolution of it.

Reply:
@PeterSchafer please check the recent changes. There is now a package `runtimeinfo` with an `interface`, implementation and some convenience functions.